content/blog/2019-10-08-remote-read-meets-streaming.md

```diff
@@ -0,0 +1,313 @@
+---
+title: Remote Read Meets Streaming
+created_at: 2019-10-08
+kind: article
+author_name: Bartlomiej Plotka
+---
+
+The new Prometheus version 2.13.0 is available and as always, it includes many fixes and improvements. You can read what's changed [here](https://github.com/prometheus/prometheus/blob/release-2.13/CHANGELOG.md). 
+However, there is one feature that some projects and users were waiting for: [chunked, streamed version of remote read API](https://docs.google.com/document/d/1JqrU3NjM9HoGLSTPYOvR217f5HBKBiJTqikEB9UiJL0/edit#heading=h.3en2gbeew2sa).  
+
+In this article I would like to present a deep dive of what we changed in the remote protocol, why it was changed and how to use it effectively.
+
+## Remote APIs
+
+Since version 1.x, Prometheus has the ability to interact directly with its storage using [remote API](https://prometheus.io/docs/prometheus/latest/storage/#remote-storage-integrations). 
+
+This API allows 3rd party systems to interact with metrics data through two methods:
+
+* **Write** - receive samples pushed by Prometheus
+* **Read** - pull samples from Prometheus
+
+![Remote read and write architecture](/assets/blog/2019-10-08/remote_integrations.png)
+
+Both methods are using HTTP with messages encoded with [protobufs](https://github.com/protocolbuffers/protobuf). 
+Both request and response for both methods are compressed using [snappy](https://github.com/google/snappy).
```

Comment:
The requests and responses

This avoids the double both

Reply:
Triple `both` kind of ;p